src/protocols/rdp/channels/audio-input/audio-buffer.c

```diff
@@ -233,6 +239,96 @@ static int guac_rdp_audio_buffer_read_sample(
 void guac_rdp_audio_buffer_write(guac_rdp_audio_buffer* audio_buffer,
         char* buffer, int length) {
 
+    pthread_mutex_lock(&(audio_buffer->lock));
+
+    /* Ignore packet if there is no buffer */
+    if (audio_buffer->packet_size == 0 || audio_buffer->packet == NULL) {
+        pthread_mutex_unlock(&(audio_buffer->lock));
+        return;
+    }
+
+    /* Allocate the new buffer. */
+    audio_stream_list *tmp = (audio_stream_list*)malloc(sizeof(audio_stream_list));
+    tmp->data = malloc(length);
+
+    /* Write the stream data to the new buffer. */
+    tmp->length = length;
+    tmp->next = NULL;
+    memcpy(tmp->data, buffer, tmp->length);
+    
+    /* Add the new buffer into the list. */
+    if (audio_buffer->first_stream_list == NULL) {
+        audio_buffer->first_stream_list = tmp;
+        audio_buffer->last_stream_list = tmp;
+    }
+    else {
+        audio_buffer->last_stream_list->next = (void*)tmp;
+        audio_buffer->last_stream_list =
+            (audio_stream_list*)audio_buffer->last_stream_list->next;
+    }
```

Comment:
If new memory will be allocated for every received `blob` of audio data, there needs to be some reasonable upper bound on how much pending audio data will be stored.